docs/articles/documentation/extending-testcafe/creating-custom-reporter/index.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: docs
+title: Creating Custom Reporter
+permalink: /documentation/extending-testcafe/creating-custom-reporter/
+---
+# Creating Custom Reporter
+
+TestCafe has a number of [built-in reporters](/testcafe/documentation/using-testcafe/common-concepts/reporters/) to generate test reports in different formats.
+You can also create a **custom reporter** that will output information in your own format and style.
+For this purpose, you can use the [TestCafe reporter generator](https://github.com/DevExpress/generator-testcafe-reporter).
+The generator will scaffold out a reporter plugin, so that you only need to write a few lines of code.
+
+## Generating Reporter Project
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-reporter` using [npm](https://www.npmjs.com/) (we assume you have pre-installed [node.js](https://nodejs.org/)).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-reporter
+```
+
+Create a new directory and go into it.
+
+```bash
+mkdir MyReporter
+cd MyReporter
+```
+
+This directory is where the generator will place your scaffolded project files.
+
+Then run the reporter generator to create a new reporter project:
+
+```bash
+yo testcafe-reporter
+```
+
+The generator will provide optional settings to customize your reporter.
+You need to specify the reporter name, your GitHub username, the URL of your website,
+and select whether you want to implement a custom error decorator for the reporter
+(you will do this later by using the [createErrorDecorator](/testcafe/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/#createerrordecorator) method).
+
+Yeoman will automatically scaffold out your reporter, install the required dependencies,
+and pull in several useful [Gulp tasks](#workflow-tasks) for your workflow.
+
+## Implementing Reporter
+
+Since the reporter has been scaffolded out, go to the reporter directory and open the `src/index.js` file.
+
+You have to implement four methods of the [ReporterPluginHost](/testcafe/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/) object providing access to your reporter.
```

Comment:
`ReporterPluginHost` is not a public object. Reporter plugin exports the single function - reporter factory. Factory should return newly created reporter object. Object should implement 4 methods listed below.